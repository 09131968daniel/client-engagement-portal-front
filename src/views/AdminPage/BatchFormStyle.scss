.BatchForm{
    height: 40vh;
    max-width:30vw; 
    background-color:white;
    border-radius:25px;
    text-align:center;
    color:#F26925;
    border:solid;
    border-color:black; 
    border-width:1px;
    padding:20px;

}

.BatchFormSubmit{
    max-width:150px;
    color:white !important;
    background-color:#F26925 !important; 
    margin:auto; 
    margin-top:90px; 
    border-radius:15px;

}