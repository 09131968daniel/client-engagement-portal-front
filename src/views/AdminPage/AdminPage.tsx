import React from "react";
import { Col, Container, Form, Input, Row } from "reactstrap";
import { NavBar } from "../../components/NavBar/NavBar";
import { NewClientButton } from "../../components/NewClientButton/NewClientButton";
import "./BatchFormStyle.scss";

export const AdminPage: React.FC = () => {
  return (
    <>
      <Container
        style={{
          height: "100vh",
          maxWidth: "100vw",
          backgroundColor: "#B9B9BA",
        }}
      >
        <NavBar />
        <Row className="justify-content-between">
          <Col xs="auto"></Col>
          <Col xs="3" className="text-left" style={{ marginTop: "50px" }}>
            <Form className="BatchForm">
              Map Batch To Client
              <Input type="select">
                <option>Select Client</option>
                <option>Dummy Client 1</option>
                <option>Dummy Client 2</option>
              </Input>
              <br />
              <Input type="select">
                <option>Select Batch</option>
                <option>Dummy Batch 1</option>
                <option>Dummy Batch 2</option>
              </Input>
              <Input className="BatchFormSubmit" type="submit">
                Submit
              </Input>
            </Form>
          </Col>
          <Col xs="3" className="text-right" style={{ marginTop: "50px" }}>
            <Form className="BatchForm">
              Unmap Batch From Client
              <Input type="select">
                <option>Select Client</option>
                <option>Dummy Client 1</option>
                <option>Dummy Client 2</option>
              </Input>
              <br />
              <Input type="select">
                Select Batch
                <option>Select Batch</option>
                <option>Dummy Batch 3</option>
                <option>Dummy Batch 4</option>
              </Input>
              <Input className="BatchFormSubmit" type="submit">
                Submit
              </Input>
            </Form>
          </Col>
          <Col xs="auto"></Col>
        </Row>
        <Row style={{ marginTop: "20px" }}>
          <Col xs="5" />
          <Col>
            <NewClientButton />
          </Col>
          <Col xs="5" />
        </Row>
        {/* <Row>
                Footer?
            </Row> */}
      </Container>

      {/* sticky footer */}
    </>
  );
};