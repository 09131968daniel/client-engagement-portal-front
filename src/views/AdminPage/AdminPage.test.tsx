import React from 'react';
import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { AdminPage } from './AdminPage'
import { Container } from 'reactstrap';
import { store } from '../../Store'
import { Provider } from 'react-redux';
configure({ adapter: new Adapter() });

/**
   * @Test
   * Component will mount
   */

it('Component Should Mount', () => {
    const wrapper = mount(
        <Provider store={store}>
            <AdminPage />
        </Provider>
    )
})
