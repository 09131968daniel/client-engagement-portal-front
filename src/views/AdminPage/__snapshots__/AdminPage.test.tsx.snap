// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AminPage should render with given state from Redux store 1`] = `
<div
  className="container"
  style={
    Object {
      "backgroundColor": "#E3E3E3",
      "maxWidth": "100vw",
      "minHeight": "100vh",
    }
  }
>
  <div
    className="justify-content-around myNav row"
  >
    <div
      className="justify-content-start logoContainer col-auto"
    >
      <a
        href="#"
      >
        <img
          alt="revature logo"
          className="myLogo"
          id="revLogo"
          src=""
        />
      </a>
    </div>
    <div
      className="d-flex align-items-center justify-content-end auto test1 col"
    >
      <div
        className="btn-group"
        onKeyDown={[Function]}
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          aria-label={null}
          className="btn btn-secondary"
          id="navDropButton"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "border": "none",
              "margin": "10px",
            }
          }
          type="button"
        >
          <span
            className="myDropdown"
            id="usersName"
            style={
              Object {
                "backgroundColor": "white",
                "border": "none",
                "color": "#474C55",
                "margin": "5px",
              }
            }
          >
            Welcome, 
            Developer User
             â–¼
          </span>
          <span
            className="myMobileDropdown"
          >
            <div
              className="hamburger-react"
              onClick={[Function]}
              onKeyUp={[Function]}
              role="button"
              style={
                Object {
                  "cursor": "pointer",
                  "height": "48px",
                  "outline": "none",
                  "position": "relative",
                  "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                  "userSelect": "none",
                  "width": "48px",
                }
              }
              tabIndex={0}
            >
              <div
                style={
                  Object {
                    "background": "#474C55",
                    "height": "3px",
                    "left": "8px",
                    "position": "absolute",
                    "top": "13px",
                    "transform": "none",
                    "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                    "width": "32px",
                  }
                }
              />
              <div
                style={
                  Object {
                    "background": "#474C55",
                    "height": "3px",
                    "left": "8px",
                    "position": "absolute",
                    "top": "23px",
                    "transform": "none",
                    "transition": "0.2s cubic-bezier(0, 0, 0, 1)",
                    "width": "32px",
                  }
                }
              />
              <div
                style={
                  Object {
                    "background": "#474C55",
                    "height": "3px",
                    "left": "8px",
                    "position": "absolute",
                    "top": "33px",
                    "transform": "none",
                    "transition": "0.4s cubic-bezier(0, 0, 0, 1)",
                    "width": "32px",
                  }
                }
              />
            </div>
          </span>
        </button>
        <div
          aria-hidden={true}
          className="dropdown-menu dropdown-menu-right"
          role="menu"
          tabIndex="-1"
        >
          <h6
            className="dropdown-header"
            id="test-head"
            onClick={[Function]}
            tabIndex="-1"
          >
            Account Options
          </h6>
          <a
            href="/login"
          >
            <button
              className="dropdown-item"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
              type="button"
            >
              Logout
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="justify-content-center my-button-row row"
  >
    <div
      className="col-2 col-sm-3 col-lg-5"
    />
    <div
      className="col-8 col-sm-6 col-lg-2"
    >
      <button
        className="batch-form-button"
        id="map-batch"
        onClick={[Function]}
      >
        Map Batch To Client
      </button>
    </div>
    <div
      className="col-2 col-sm-3 col-lg-5"
    />
  </div>
  <div
    className="justify-content-center my-button-row row"
  >
    <div
      className="col-2 col-sm-3 col-lg-5"
    />
    <div
      className="col-8 col-sm-6 col-lg-2"
    >
      <button
        className="batch-form-button"
        id="unmap-batch"
        onClick={[Function]}
      >
        Unmap Batch From Client
      </button>
    </div>
    <div
      className="col-2 col-sm-3 col-lg-5"
    />
  </div>
  <div
    className="justify-content-between my-form-row row"
  >
    <div
      className="col-sm-1 col-md-1 col-lg-2 col-xl-3"
    />
    <div
      className="text-left col-sm-5 col-md-5 col-lg-4 col-xl-3"
      style={
        Object {
          "marginTop": "50px",
        }
      }
    >
      <form
        className="batch-form"
        onSubmit={[Function]}
      >
        <h5>
          Map Batch To Client
        </h5>
        <select
          className="form-control"
          name="mappedClientEmail"
        >
          <option
            disabled={true}
            selected={true}
          >
            Select Client
          </option>
        </select>
        <br />
        <select
          className="form-control"
          name="mappedBatchId"
        >
          <option
            disabled={true}
            selected={true}
          >
            Select Batch:
          </option>
        </select>
        <span>
          
        </span>
        <input
          className="batch-form-submit"
          id="test-submit-map"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
    <div
      className="text-right col-sm-5 col-md-5 col-lg-4 col-xl-3"
      style={
        Object {
          "marginTop": "50px",
        }
      }
    >
      <form
        className="batch-form"
        onSubmit={[Function]}
      >
        <h5>
          Unmap Batch From Client
        </h5>
        <select
          className="form-control"
          disabled={false}
          id="map-client-to-batch"
          name="unmappingClientEmail"
          onChange={[Function]}
        >
          <option
            disabled={true}
            selected={true}
          >
            Select Client
          </option>
        </select>
        <br />
        <select
          className="form-control"
          disabled={false}
          name="unmappingBatchId"
        >
          Select Batch
          <option
            disabled={true}
            selected={true}
          >
            Select Batch:
          </option>
        </select>
        <span>
          
        </span>
        <input
          className="batch-form-submit"
          disabled={false}
          id="test-submit-unmap"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
    <div
      className="col-sm-1 col-md-1 col-lg-2 col-xl-3"
    />
  </div>
  <div
    className="row"
    style={
      Object {
        "marginTop": "20px",
      }
    }
  >
    <div
      className="col-2 col-sm-3 col-md-4 col-lg-4 col-xl-5"
    />
    <div
      className="col-8 col-sm-6 col-md-4 col-lg-4 col-xl-2"
    >
      <button
        aria-label={null}
        className="create-account-button btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Create Account
      </button>
    </div>
    <div
      className="col-2 col-sm-3 col-md-4 col-lg-4 col-xl-5"
    />
  </div>
</div>
`;
